features/alarm-change-notifier/main-module/src/main/java/org/opennms/plugins/dbnotifier/DbNotifierDataSourceFactory.java

```diff
@@ -0,0 +1,211 @@
+ * field values rather than loading from opennms-datasources.xml
+ */
+public class DbNotifierDataSourceFactory {
+	private static final Logger LOG = LoggerFactory.getLogger(DbNotifierDataSourceFactory.class);
+
+	private static final String OPENNMS_DATASOURCE_CONFIG_FILE_NAME="opennms-datasources.xml";
+
+	private static final String OPENNMS_DATA_SOURCE_NAME = "opennms";
+
+	private String userName  = null;
+	private String passWord  = null;
+	private String dataBaseName  = null;
+	private String hostname=null;
+	private int port=5432;
+	private String dsfileUri=null;
+
+
+	public String getUserName() {
+		return userName;
+	}
+
+	public void setUserName(String userName) {
+		this.userName = userName;
+	}
+
+	public String getPassWord() {
+		return passWord;
+	}
+
+	public void setPassWord(String passWord) {
+		this.passWord = passWord;
+	}
+
+	public String getDataBaseName() {
+		return dataBaseName;
+	}
+
+	public void setDataBaseName(String dataBaseName) {
+		this.dataBaseName = dataBaseName;
+	}
+
+	public String getHostname() {
+		return hostname;
+	}
+
+	public void setHostname(String hostname) {
+		this.hostname = hostname;
+	}
+
+	public String getPort() {
+		return Integer.toString(port);
+	}
+
+	public void setPort(String port) {
+		this.port = Integer.parseInt(port);
+	}
+
+	/**
+	 * @return the dsfileUri
+	 */
+	public String getDataSourceFileUri() {
+		return dsfileUri;
+	}
+
+	/**
+	 * @param dsfileUri the dsfileUri to set
+	 */
+	public void setDataSourceFileUri(String fileUri) {
+		this.dsfileUri = fileUri;
+	}
+
+	public PGDataSource getPGDataSource(){
+
+		PGDataSource pgdc=new PGDataSource();
+		pgdc.setHost(hostname);
+		pgdc.setPort(port);
+		pgdc.setDatabase(dataBaseName);
+		pgdc.setUser(userName);
+		pgdc.setPassword(passWord);
+
+		return pgdc;
+
+	}
+
+
+	/**
+	 *  This method initialises the class by loading the database values
+	 *  either from the config .cfg file or from opennms-datasources.xml
+	 */
+	public void init(){
+
+		if(dataBaseName!=null){
+			LOG.debug("using values supplied in .cfg file for host: "+hostname
+					+ " port: "+port
+					+ " dataBaseName "+ dataBaseName+" userName: "+userName+ "  password :" + passWord);
+		} else {
+
+			try {
+
+				// try loading from file or from classpath
+				InputStream istream=null;
+				if(dsfileUri!=null){
+					File dsFile=new File(dsfileUri);
+					LOG.debug("loading database config from "+dsFile.getAbsolutePath());
+					istream= new FileInputStream(dsFile);
+				}
+
+				if(istream==null){	
+					LOG.debug("loading database config from classpath file"+OPENNMS_DATASOURCE_CONFIG_FILE_NAME);
+					istream= this.getClass().getClassLoader().getResourceAsStream(OPENNMS_DATASOURCE_CONFIG_FILE_NAME);
+					if(istream==null) throw new RuntimeException("could not load database config from classpath "+OPENNMS_DATASOURCE_CONFIG_FILE_NAME);
+				}
+
+
+				DocumentBuilderFactory factory = DocumentBuilderFactory.newInstance();
```

Comment:
I looked at the existing dao but there way no way to get access to all  the raw database information in order to construct the URI needed to set up jdbc-ng.
I needed to get access to the raw definitions of the opennms database hostname, port, database name, username and password to access postgres. This is used to construct a different data source url for use with jdbc-ng. The only way to get that information was to parse the raw opennms-datasources.xml file. It is not accessible from the dao (no suitable get methods on the data source)